File: 240410/십진수와 이진수 2/decimal-and-binary-number-2.py
bin_number = input()

num = 0
for idx in range(len(bin_number)):
    num = num * 2 + int(bin_number[idx])


num *= 17
digits = []
while True:
    if num < 2:
        digits.append(num)
        break

    digits.append(num % 2)
    num //= 2

print("".join(map(str, digits[::-1])))

File: 260115/십진수와 이진수 2/decimal-and-binary-number-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    string bin_number;
    cin >> bin_number;

    // Please write your code here.
    long long num = 0;
    for(char ch: bin_number)
    {
        num = num * 2 + (ch - '0');
    }
    num *= 17;

    if (num == 0) 
    {
        cout << 0;
        return 0;
    }

    string digits;
    while (num > 0) 
    {
        digits.push_back(char('0' + (num % 2)));
        num /= 2;
    }
    reverse(digits.begin(), digits.end());

    cout << digits;
    return 0;
}